blaze/expr/collections.py

```diff
@@ -831,7 +1014,7 @@ def shift(expr, n):
     return Shift(expr, n)
 
 
-dshape_method_list.extend([(iscollection, set([sort, head, tail, sample])),
+dshape_method_list.extend([(iscollection, set([asc, desc, sort, sort2, head, tail, sample])),
```

Comment:
There may be another problem with desc(). With the existing API it is simple to sort by all columns in descending order:

```python
str(bz.compute(data.sort(ascending=False)))
'SELECT numeric_table2.id, numeric_table2.alpha, numeric_table2.beta, numeric_table2.gamma, numeric_table2.delta \nFROM numeric_table2 ORDER BY numeric_table2.id DESC, numeric_table2.alpha DESC, numeric_table2.beta DESC, numeric_table2.gamma DESC, numeric_table2.delta DESC'
```

Now all columns have to be enumerated -- I think that was part of the reason why the first patch allowed ```data.desc()```.